.github/workflows/test_on_push.yml

```diff
@@ -351,11 +338,6 @@ jobs:
           cache: 'pip'
           cache-dependency-path: setup.py
 
-      - name: Install PyBaMM dependencies
-        run: |
-          pip install --upgrade pip wheel setuptools nox
-          pip install -e .[all,docs]
-
```

Comment:
Now that we have separate jobs for different sets of tests, this step turns out to be redundant (we can save ~1-2 minutes per platform by removing it) because `nox` installs all dependencies in a virtual environment again anyway